EngineLayer/ModernSearch/ModernSearchEngine.cs

```diff
@@ -39,84 +40,40 @@ protected override MetaMorpheusEngineResults RunSpecific()
         {
             double progress = 0;
             int oldPercentProgress = 0;
-            ReportProgress(new ProgressEventArgs(oldPercentProgress, "Performing modern search... " + CurrentPartition + "/" + CommonParameters.TotalPartitions, NestedIds));
+            ReportProgress(new ProgressEventArgs(oldPercentProgress, "Performing modern search... " +
+                CurrentPartition + "/" + CommonParameters.TotalPartitions, NestedIds));
 
             byte byteScoreCutoff = (byte)CommonParameters.ScoreCutoff;
 
-            int maxThreadsPerFile = CommonParameters.MaxThreadsToUsePerFile;
-            int[] threads = Enumerable.Range(0, maxThreadsPerFile).ToArray();
-            Parallel.ForEach(threads, (i) =>
-            {
-                byte[] scoringTable = new byte[PeptideIndex.Count];
-                List<int> idsOfPeptidesPossiblyObserved = new List<int>();
-
-                for (; i < ListOfSortedMs2Scans.Length; i += maxThreadsPerFile)
+            Parallel.ForEach(Partitioner.Create(0, ListOfSortedMs2Scans.Length),
+                new ParallelOptions { MaxDegreeOfParallelism = CommonParameters.MaxThreadsToUsePerFile },
+                (range, loopState) =>
                 {
-                    // Stop loop if canceled
-                    if (GlobalVariables.StopLoops)
-                    {
-                        return;
-                    }
-
-                    // empty the scoring table to score the new scan (conserves memory compared to allocating a new array)
-                    Array.Clear(scoringTable, 0, scoringTable.Length);
-                    idsOfPeptidesPossiblyObserved.Clear();
-                    Ms2ScanWithSpecificMass scan = ListOfSortedMs2Scans[i];
-
-                    // get fragment bins for this scan
-                    List<int> allBinsToSearch = GetBinsToSearch(scan, FragmentIndex, CommonParameters.DissociationType);
-
-                    // get allowed theoretical masses from the known experimental mass
```

Comment:
I still like this comment to help describe what we're doing for the indexed search.